packages/react-aria-components/src/Checkbox.tsx

```diff
@@ -160,9 +165,15 @@ function CheckboxGroup(props: CheckboxGroupProps, ref: ForwardedRef<HTMLDivEleme
 export const CheckboxContext = createContext<ContextValue<CheckboxProps, HTMLLabelElement>>(null);
 
 function Checkbox(props: CheckboxProps, ref: ForwardedRef<HTMLLabelElement>) {
-  [props, ref] = useContextProps(props, ref, CheckboxContext);
-  let inputRef = useRef<HTMLInputElement>(null);
+  let {
+    inputRef: userProvidedInputRef = null,
+    ...otherProps
+  } = props;
+  [props, ref] = useContextProps(otherProps, ref, CheckboxContext);
   let groupState = useContext(CheckboxGroupStateContext);
+  let defaultInputRef = useRef<HTMLInputElement>(null);
+  let userInputRef = useObjectRef(mergeRefs(userProvidedInputRef, props.inputRef !== undefined ? props.inputRef : null));
+  let inputRef = userInputRef || defaultInputRef;
```

Comment:
Small nit, but I believe useObjectRef always returns a value so you can get rid of `defaultInputRef` I believe.

```suggestion
  let inputRef = useObjectRef(mergeRefs(userProvidedInputRef, props.inputRef));
```

Reply:
ah yep i think you're right, will remove!